GVFS/GVFS.Common/Git/GitProcess.cs

```diff
@@ -674,15 +685,16 @@ private Result InvokeGitOutsideEnlistment(string command)
         /// Invokes git.exe against an enlistment's .git folder.
         /// This method should be used only with git-commands that ignore the working directory
         /// </summary>
-        private Result InvokeGitAgainstDotGitFolder(string command)
+        private Result InvokeGitAgainstDotGitFolder(string command, string gitObjectsDirectory = null)

```

Comment:
If we are going to start adding optional parameters, can we cut out this middle function and just add optional parameters to the called function?

Reply:
I thought this exact same, but then looking at all the options we default decided against it:

                command,
                workingDirectory: Environment.SystemDirectory,
                dotGitDirectory: this.dotGitRoot,
                useReadObjectHook: false,
                writeStdIn: writeStdIn,
                parseStdOutLine: parseStdOutLine,
                timeoutMs: -1,
                gitObjectsDirectory: gitObjectsDirectory);

That's a lot to call directly and hope defaults don't change.  Maybe it's fine?